assets/app/scripts/services/applicationGenerator.js

```diff
@@ -59,13 +34,32 @@ angular.module("openshiftConsole")
               protocol: parts[1].toUpperCase()
             });
         });
+
+        // Since the exposed ports in Docker image metadata are not in any
+        // order, sort the ports from lowest to highest.
+        ports.sort(function(left, right) {
+          return left.containerPort - right.containerPort;
```

Comment:
I don't know this well enough to know... will these always exist and be numeric?